server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1589,6 +1509,148 @@ static String resolveExpression(String expression, LongSupplier getTime) {
         }
     }
 
+    public static final class ExplicitResourceNameFilter {
+
+        private ExplicitResourceNameFilter() {
+            // Utility class
+        }
+
+        public static List<String> filter(Context context, List<String> expressions) {
+            List<String> result = new ArrayList<>(expressions.size());
+            for (ExpressionIterable.Expression expression : new ExpressionIterable(context, expressions)) {
+                validateAliasOrIndex(expression);
+                if (expression.isWildcard() || expression.isExclusion() || ensureAliasOrIndexExists(context, expression.toString())) {
+                    result.add(expression.isExclusion() ? "-" + expression : expression.toString());
```

Comment:
The "explicit resource name" filter is only concerned with non-wildcards and non-exclusions.
For such names, if `ignore_unavailable` is `true` then it silently removes them (this is modification); otherwise, if `false` throws a "not found" kind of exception (this is existing behavior).